src/Integration/State/StateManager.cs

```diff
@@ -87,12 +89,16 @@ public void SetBoundProject(ProjectInformation project)
             ProjectViewModel projectViewModel = this.ManagedState.ConnectedServers.SelectMany(s => s.Projects).SingleOrDefault(p => p.ProjectInformation == project);
             Debug.Assert(projectViewModel != null, "Expecting a single project mapped to project information");
             this.ManagedState.SetBoundProject(projectViewModel);
+
+            this.BindingStateChanged?.Invoke(this, null);
```

Comment:
Also, a proper `EventArgs` should be passed. That might simplify the consumer side.